### PR TITLE
Change widget inits to Fliplet.Widget.instance

### PR DESCRIPTION
**Pull request by YaroslavOvdii**
@sofiiakvasnevska 
## Issue
https://github.com/Fliplet/fliplet-studio/issues/5354

## Description
Change widget inits to Fliplet.Widget.instance

## Backward compatibility

This change is fully backward compatible.

### CODE REVIEW

File: js/build.js [hunk 1]
```diff
@@ -1,9 +1,7 @@
-$('[data-login-ds-id]').each(function() {
-  var _this = this;
+Fliplet.Widget.instance('login-ds', function (instanceData) {
```
Comment (tonytlwu) [inline js/build.js]:
```suggestion
Fliplet.Widget.instance('login-ds', function (data) {
```

File: js/build.js [hunk 2]
```diff
@@ -1,9 +1,7 @@
...
   var $container = $(this);
-  var widgetId = $container.attr('data-login-ds-id');
-  var containerSelector = '[data-login-ds-id="'+widgetId+'"]';
-  var widgetUuid = $container.attr('data-login-ds-uuid');
+  var widgetId = instanceData.id;
+  var widgetUuid = instanceData.uuid;
   var data = Fliplet.Widget.getData(widgetId);
```
Comment (tonytlwu) [inline js/build.js]:
```suggestion
```

File: js/build.js [hunk 3]
```diff
@@ -13,6 +11,7 @@ $('[data-login-ds-id]').each(function() {
   }
 
   this.pvName = 'login_data_source_component_' + Fliplet.Env.get('appId');
+  // Unused object should we remote it?
```
Comment (tonytlwu) [inline js/build.js]: If it's unused, remove it.

File: js/build.js [hunk 4]
```diff
@@ -110,19 +109,24 @@ $('[data-login-ds-id]').each(function() {
...
         return Promise.reject(error);
       });
   }
 
   function resetFromDataSource(data_source_id, where) {
-    return Fliplet.DataSources.connect(data.dataSource, { offline: false })
-      .then(function(dataSource) {
-        return dataSource.sendValidation({type: 'email', where: where})
+    return Fliplet.DataSources.connect(data.dataSource, {
+        offline: false
```
Comment (tonytlwu) [inline js/build.js]:
```suggestion
      offline: false
```

File: js/build.js [hunk 5]
```diff
@@ -110,19 +109,24 @@ $('[data-login-ds-id]').each(function() {
...
+      })
```
Comment (tonytlwu) [inline js/build.js]:
```suggestion
    })
```

File: js/build.js [hunk 6]
```diff
@@ -315,83 +317,90 @@ $('[data-login-ds-id]').each(function() {
...
+          };
           where[data.emailColumn] = resetEmail;
 
           Fliplet.Session.get()
-            .then(function() {
-              dataSource.validate({ type: 'email', where: where })
-                .then(function(entry) {
+            .then(function () {
+              dataSource.validate({
+                  type: 'email',
```
Comment (tonytlwu) [inline js/build.js]:
```suggestion
                type: 'email',
```

File: js/build.js [hunk 7]
```diff
@@ -315,83 +317,90 @@ $('[data-login-ds-id]').each(function() {
...
+                  where: where
```
Comment (tonytlwu) [inline js/build.js]:
```suggestion
                where: where
```

File: js/build.js [hunk 8]
```diff
@@ -315,83 +317,90 @@ $('[data-login-ds-id]').each(function() {
...
+                })
```
Comment (tonytlwu) [inline js/build.js]:
```suggestion
              })
```

File: js/build.js [hunk 9]
```diff
@@ -520,58 +535,63 @@ $('[data-login-ds-id]').each(function() {
...
+      $container.find('.state[data-state=verify-code] .form-group').removeClass('has-error');
+      $container.find('.resend-code').addClass('hidden');
 
-      calculateElHeight($(containerSelector).find('.state[data-state=verify-code]'));
+      calculateElHeight($container.find('.state[data-state=verify-code]'));
 
-      Fliplet.DataSources.connect(data.dataSource, { offline: false })
-        .then(function(dataSource) {
+      Fliplet.DataSources.connect(data.dataSource, {
+          offline: false
```
Comment (tonytlwu) [inline js/build.js]:
```suggestion
        offline: false
```

File: js/build.js [hunk 10]
```diff
@@ -520,58 +535,63 @@ $('[data-login-ds-id]').each(function() {
...
+        })
```
Comment (tonytlwu) [inline js/build.js]:
```suggestion
      })
```

File: js/build.js [hunk 11]
```diff
@@ -520,58 +535,63 @@ $('[data-login-ds-id]').each(function() {
...
+        .then(function (dataSource) {
           var where = {};
           where[data.emailColumn] = resetEmail;
-          dataSource.sendValidation({ type: 'email', where: where })
+          dataSource.sendValidation({
+              type: 'email',
```
Comment (tonytlwu) [inline js/build.js]:
```suggestion
            type: 'email',
```

File: js/build.js [hunk 12]
```diff
@@ -520,58 +535,63 @@ $('[data-login-ds-id]').each(function() {
...
+              where: where
```
Comment (tonytlwu) [inline js/build.js]:
```suggestion
            where: where
```

File: js/build.js [hunk 13]
```diff
@@ -520,58 +535,63 @@ $('[data-login-ds-id]').each(function() {
...
+            })
```
Comment (tonytlwu) [inline js/build.js]:
```suggestion
          })
```

File: js/build.js [hunk 14]
```diff
@@ -110,20 +99,25 @@ $('[data-login-ds-id]').each(function() {
 
   function loginFromDataSource(data_source_id, where) {
     return Fliplet.Session.authorize({
-      passport: 'dataSource',
-      dataSourceId: data_source_id,
-      where: where
-    })
+        passport: 'dataSource',
```
Comment (tonytlwu) [inline js/build.js]:
```suggestion
      passport: 'dataSource',
```

File: js/build.js [hunk 15]
```diff
@@ -520,60 +525,65 @@ $('[data-login-ds-id]').each(function() {
...
+      $container.find('.resend-code').addClass('hidden');
+
+      calculateElHeight($container.find('.state[data-state=verify-code]'));
+
+      Fliplet.DataSources.connect(data.dataSource, {
+        offline: false
+      })
+      .then(function (dataSource) {
+        var where = {};
+        where[data.emailColumn] = resetEmail;
```
Comment (tonytlwu) [inline js/build.js]:
```suggestion

        where[data.emailColumn] = resetEmail;
```

File: js/build.js [hunk 16]
```diff
@@ -520,60 +525,65 @@ $('[data-login-ds-id]').each(function() {
...
+        dataSource.sendValidation({
+          type: 'email',
+          where: where
+        })
+        .then(function () {
+          $container.find('.pin-code-field').val('');
+          $container.find('.pin-sent-success').removeClass('hidden');
+          if ($container.find('.state[data-state=verify-code] .form-group').hasClass('has-error')) {
```
Comment (tonytlwu) [inline js/build.js]:
```suggestion

          if ($container.find('.state[data-state=verify-code] .form-group').hasClass('has-error')) {
```

File: js/build.js [hunk 17]
```diff
@@ -520,60 +525,65 @@ $('[data-login-ds-id]').each(function() {
...
+            $container.find('.state[data-state=verify-code] .form-group').removeClass('has-error');
+          }
+          if (!$container.find('.resend-code').hasClass('hidden')) {
```
Comment (tonytlwu) [inline js/build.js]:
```suggestion

          if (!$container.find('.resend-code').hasClass('hidden')) {
```

File: js/build.js [hunk 18]
```diff
@@ -110,20 +99,25 @@ $('[data-login-ds-id]').each(function() {
...
   function loginFromDataSource(data_source_id, where) {
     return Fliplet.Session.authorize({
-      passport: 'dataSource',
-      dataSourceId: data_source_id,
-      where: where
-    })
+        passport: 'dataSource',
+        dataSourceId: data_source_id,
+        where: where
+      })
```
Comment (tonytlwu) [inline js/build.js]:
```suggestion
    })
```

File: js/build.js [hunk 19]
```diff
@@ -114,16 +103,21 @@ $('[data-login-ds-id]').each(function() {
...
 
   function resetFromDataSource(data_source_id, where) {
-    return Fliplet.DataSources.connect(data.dataSource, { offline: false })
-      .then(function(dataSource) {
-        return dataSource.sendValidation({type: 'email', where: where})
-      });
+    return Fliplet.DataSources.connect(data.dataSource, {
+      offline: false
+    })
+    .then(function (dataSource) {
```
Comment (tonytlwu) [inline js/build.js]:
```suggestion
      .then(function (dataSource) {
```

File: js/build.js [hunk 20]
```diff
@@ -114,16 +103,21 @@ $('[data-login-ds-id]').each(function() {
       dataSourceId: data_source_id,
       where: where
     })
-      .catch(function (error) {
-        return Promise.reject(error);
-      });
+    .catch(function (error) {
```
Comment (tonytlwu) [inline js/build.js]: You chained the `.then()` in the same line as the function. You shoudl either move the `.then()` to a new line, or move the `.catch()` to the previous line -- for consistency.

File: js/build.js [hunk 21]
```diff
@@ -114,16 +103,21 @@ $('[data-login-ds-id]').each(function() {
...
-      });
+    return Fliplet.DataSources.connect(data.dataSource, {
+      offline: false
+    })
+      .then(function (dataSource) {
+        return dataSource.sendValidation({
+          type: 'email',
+          where: where
+        })
+    });
```
Comment (tonytlwu) [inline js/build.js]:
```suggestion
      });
```

...or move `.then()` to the previous line

**Comment by YaroslavOvdii** [comment]
Created new [PR ](https://github.com/Fliplet/fliplet-widget-login-data-source/pull/67) with already added [eslint](https://github.com/Fliplet/fliplet-widget-login-data-source/pull/64) rules.

**Pull request closed without merging**